### PR TITLE
chore: linting

### PR DESCRIPTION
**Pull request by faddat**
## Description

Problem: Golangci-lint masked real issues because it fired only on diffs.

Soulution: don't fire the linter only on diffs, lint, then merge after linting. 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: price-feeder/go.mod [hunk 1]
```diff
@@ -41,7 +41,7 @@ require (
 	github.com/ChainSafe/go-schnorrkel v0.0.0-20200405005733-88cbf1b4c40d // indirect
 	github.com/Djarvur/go-err113 v0.0.0-20210108212216-aea10b59be24 // indirect
 	github.com/GaijinEntertainment/go-exhaustruct/v2 v2.3.0 // indirect
-	github.com/Gravity-Bridge/Gravity-Bridge/module v1.5.3 // indirect
+	github.com/Gravity-Bridge/Gravity-Bridge/module v1.7.2 // indirect
```
Comment (toteki) [inline price-feeder/go.mod:44]: Note: This change doesn't seem to be a purely linting. Was it related?

Comment (faddat) [reply price-feeder/go.mod:44]: yes.  WIthout it, we couldn't use the go workspaces. 

We still use our fork.

Comment (faddat) [reply price-feeder/go.mod:44]: FYI, this branch is where all the action is right now.  

WIll do more with price feed tomorrow.

Comment (faddat) [reply price-feeder/go.mod:44]: viz:
https://github.com/umee-network/umee/blob/1309241296d46bd91a19957868044245136bf8e7/price-feeder/go.mod#L305

actual used code is unchanged.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -1,46 +1,30 @@
-name: Lint
```
Comment (robert-zaremba) [inline .github/workflows/lint.yml:1]: let's don't change the names, we use them for required actions check.

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -1,46 +1,30 @@
...
+name: golangci-lint
 on:
-  pull_request:
   push:
+    tags:
+      - v*
     branches:
+      - master
```
Comment (robert-zaremba) [inline .github/workflows/lint.yml:7]: there is no `master` branch`

File: go.mod [hunk 4]
```diff
@@ -5,7 +5,7 @@ go 1.19
 require (
 	cosmossdk.io/errors v1.0.0-beta.7
 	cosmossdk.io/math v1.0.0-beta.3
-	github.com/Gravity-Bridge/Gravity-Bridge/module v1.5.3
+	github.com/Gravity-Bridge/Gravity-Bridge/module v1.7.2
```
Comment (robert-zaremba) [inline go.mod:8]: Why changing this? It will be a non-compatible, breaking change. Please roll back.

File: go.work [hunk 5]
```diff
@@ -0,0 +1,3 @@
+go 1.19
+
+use ./price-feeder
```
Comment (robert-zaremba) [inline go.work:3]: let's remove this - we have old PR for this, and it's not related to this changset.

File: price-feeder/go.mod (hunk 1 above)
Comment (robert-zaremba) [reply price-feeder/go.mod:44]: As noted in comment above, please rollback. Also, we have other old PR for using go workspaces, and this is not relevant to this PR (linting update)

File: price-feeder/oracle/provider/binance_test.go [hunk 6]
```diff
@@ -10,6 +10,16 @@ import (
...
 
+// these constants are used during testing.
+const (
+	volume        = "2396974.02000000"
+	lastPriceAtom = "34.69000000"
+	lastPrice     = "34.69000000"
+	lastPriceLuna = "41.35000000"
+	price         = "34.689998626708984000"
+	lastPriceUmee = "41.35000000"
+)
```
Comment (robert-zaremba) [inline price-feeder/oracle/provider/binance_test.go:21]: :+1:

**Review by robert-zaremba — COMMENTED** [review]
Let's make this PR only linter oriented.

File: price-feeder/go.mod (hunk 1 above)
Comment (faddat) [reply price-feeder/go.mod:44]: Hmmmmmmmmm.....
Ok I will pull that commit.

File: .github/workflows/lint.yml (hunk 3 above)
Comment (faddat) [reply .github/workflows/lint.yml:7]: These are the defaults I'll fix that up

File: go.work (hunk 5 above)
Comment (faddat) [reply go.work:3]: Okay

File: go.mod (hunk 4 above)
Comment (faddat) [reply go.mod:8]: it doesn't change the code run.  But no worries, will change.

**Comment by faddat** [comment]
I'll add a changelog to this

File: price-feeder/go.mod (hunk 1 above)
Comment (robert-zaremba) [reply price-feeder/go.mod:44]: yes, and not mangle in this PR things not related to linting (eg go workspaces, and a wrong gravity bridge version)

Comment (robert-zaremba) [reply price-feeder/go.mod:44]: up, let's undo go.mod for gravity bridge.

File: go.mod (hunk 4 above)
Comment (robert-zaremba) [reply go.mod:8]: up

**Comment by toteki** [comment]
#1649 should have most of the linting covered (but not the changes to CI)

Feel free to close (or open an issue if you still have unaddressed concerns)

**Pull request closed without merging**